packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts

```diff
@@ -730,4 +732,60 @@ export class PluginLedgerConnectorBesu
     );
     return { transaction };
   }
+  public async getBesuRecord(
+    request: BesuRecordLocatorV1Request,
+  ): Promise<BesuRecordLocatorV1Response> {
+    const fnTag = `${this.className}#getBesuRecord()`;
+    this.log.info(request);
+    //////////////////////////////////////////////
+    let abi: any = undefined;
+    const resp: BesuRecordLocatorV1Response = {};
+    const trans = request.transactionHash;
+
+    if (trans != undefined) {
+      const transaction = await this.web3.eth.getTransaction(trans);
+      if (transaction.input) {
+        resp.ledgerId = this.instanceId;
+        resp.transactionInputData = transaction.input;
+        return resp;
+      }
+    }
+
+    if (request.invokeCall != undefined) {
+      if (request.invokeCall.contractAbi != undefined) {
+        if (typeof request.invokeCall.contractAbi === "string") {
+          abi = JSON.parse(request.invokeCall.contractAbi);
+        } else {
+          abi = request.invokeCall.contractAbi;
+        }
+      }
+      const { contractAddress } = request.invokeCall;
+      const contractInstance = new this.web3.eth.Contract(abi, contractAddress);
+      const methodRef = contractInstance.methods[request.invokeCall.methodName];
+      Checks.truthy(
+        methodRef,
+        `${fnTag} YourContract.${request.invokeCall.methodName}`,
+      );
+      const method: ContractSendMethod = methodRef(
+        ...request.invokeCall.params,
+      );
+
+      if (
+        request.invokeCall.invocationType === EthContractInvocationType.Call
+      ) {
+        const callOutput = await (method as any).call();
+        const res: BesuRecordLocatorV1Response = {
+          ledgerId: this.instanceId,
```

Comment:
@AzaharaC Same as above regarding ledgerId !== instanceId

Reply:
@AzaharaC Please remove the ledgerId, assigning the instanceId to it could mislead people.

Reply:
@AzaharaC Please remove the ledgerId, assigning the instanceId to it could mislead people.